### ISSUE TITLE
CasperJS & Grunt fails

### ISSUE THREAD

**Issue by robin-obrien** [casperjs/casperjs#1451]
Hi,

I am running Grunt for running tasks of CasperJS. No matter what I try it fails after around an 10 minutes, mostly with the following errors. When I contacted the Grunt project they said it is related to a failure at CasperJS. These are usually the last messages I see when running grunt with the options "--verbose --debug --stack --force".
What might be the issue the hangs up the task?

```
[debug] [phantom] url changed to "https://www.somesite.com"
>> Exited with code: 1.
Warning: Task "exec:xyz" failed. Used --force, continuing.

Done, but with warnings. 
```

```
[debug] [phantom] opening url: www.somesite.com, HTTP GET
[debug] [phantom] Navigation requested: url=www.somesite.com, type=Other, willNavigate=true, isMainFrame=true
>> Exited with code: 1.
Warning: Task "exec:xyz" failed. Used --force, continuing.

Done, but with warnings.
```

When running without the option --force, I get the following trace:

```
Warning: Task "exec:xyz" failed. Use --force to continue.
Error: Task "exec:xyz1" failed.
    at Task.<anonymous> (project/node_modules/grunt/lib/util/task.js:205:15)
    at null._onTimeout (project/node_modules/grunt/lib/util/task.js:241:33)
    at Timer.listOnTimeout (timers.js:92:15)
```

My node version is v4.3.1, grunt version grunt-cli: v0.1.13, grunt: v0.4.5, casperjs: 1.1.0-beta5

**Comment by robin-obrien** [comment]
When running without the option --force, I get the following trace:

```
Warning: Task "exec:xyz" failed. Use --force to continue.
Error: Task "exec:xyz1" failed.
    at Task.<anonymous> (project/node_modules/grunt/lib/util/task.js:205:15)
    at null._onTimeout (project/node_modules/grunt/lib/util/task.js:241:33)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by istr** [comment]
> it fails after around an 10 minutes

Maybe a simple OOM? Could it be that your CasperJS / PhantomJS instances are not discarded by the grunt task?

**Comment by robin-obrien** [comment]
I suspect that it is not the case as the I monitor the memory during the run and it doesn't seem to reach its limit. 
Any other idea would be very much appreciated

**Comment by robin-obrien** [comment]
The feedback from grunt was:

> The failure is not at Grunt's fault and Grunt itself doesn't have a timeout on tasks. If running casperjs without grunt-exec works, I recommend looking into why spawning casperjs as a child process in Node.js in the way that grunt-exec does might cause an issue. Or perhaps casperjs is succeeding but failing to exit the process in way grunt-exec deems successful. There could be a number of reasons for casperjs to fail or timeout.
> 
> In either case, this isn't an issue with Grunt.

**Comment by kensoh** [comment]
@robin-obrien bumping this up, not sure if this issue is still happening? Thought that Grunt is popular tool, will be a pity if it doesn't run well with CasperJS.

**Comment by kensoh** [comment]
Hi @robin-obrien not sure if this issue is resolved for you with latest CasperJS v1.1.4 and PhantomJS v2.1.1? Closing this for now but please reopen if it is still happening. Thanks for raising this!
